### DISCUSSION TITLE
Noob needs help getting things going.

### DISCUSSION THREAD

**Discussion by N6REJ** [JanDeDobbeleer/oh-my-posh#2687] [Troubleshoot]
Pardon me if I'm being really dumb but I'm trying to get this going and having all kinds of issues.
I've managed to get oh-my-posh installed using windows store and when I run 
```oh-my-posh font install agave``` it stalls and then I get an error
```
Caught panic:

runtime error: invalid memory address or nil pointer dereference

Restoring terminal...

goroutine 1 [running]:
runtime/debug.Stack()
        /opt/hostedtoolcache/go/1.18.4/x64/src/runtime/debug/stack.go:24 +0x65
runtime/debug.PrintStack()
        /opt/hostedtoolcache/go/1.18.4/x64/src/runtime/debug/stack.go:16 +0x19
github.com/charmbracelet/bubbletea.(*Program).StartReturningModel.func3()
        /home/runner/go/pkg/mod/github.com/charmbracelet/bubbletea@v0.22.0/tea.go:371 +0x95
panic({0x108f660, 0x1c195b0})
        /opt/hostedtoolcache/go/1.18.4/x64/src/runtime/panic.go:844 +0x258
oh-my-posh/font.(*main).Update(0xc000096100, {0x10e2c60?, 0xc0001f0360?})
        /home/runner/work/oh-my-posh/oh-my-posh/src/font/cli.go:171 +0x281
github.com/charmbracelet/bubbletea.(*Program).StartReturningModel(0xc0000be000)
        /home/runner/go/pkg/mod/github.com/charmbracelet/bubbletea@v0.22.0/tea.go:540 +0x1291
github.com/charmbracelet/bubbletea.(*Program).Start(...)
        /home/runner/go/pkg/mod/github.com/charmbracelet/bubbletea@v0.22.0/tea.go:549
oh-my-posh/font.Run({0xc0000340c0, 0x5})
        /home/runner/work/oh-my-posh/oh-my-posh/src/font/cli.go:239 +0x94
oh-my-posh/cli.glob..func7(0x1c25860?, {0xc0000921e0?, 0x2?, 0x2?})
        /home/runner/work/oh-my-posh/oh-my-posh/src/cli/font.go:35 +0x7f
github.com/spf13/cobra.(*Command).execute(0x1c25860, {0xc000092180, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:876 +0x67b
github.com/spf13/cobra.(*Command).ExecuteC(0x1c25360)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
oh-my-posh/cli.Execute({0x1288734?, 0xc000062000?})
        /home/runner/work/oh-my-posh/oh-my-posh/src/cli/root.go:40 +0x4a
main.main()
        /home/runner/work/oh-my-posh/oh-my-posh/src/main.go:10 +0x27
PS C:\windows\system32>
```
What I IDEALLY would like to happen short term is 
oh-my-posh uses [Paradox theme](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/themes/paradox.omp.json)
![image](https://user-images.githubusercontent.com/1850089/185806627-d5e77480-b359-47c4-8aaa-dad2e98baf8b.png)
with agave font in "windows terminal" and/or consolez

LONG TERM: 
I'd like to replace our [module_consolez](https://github.com/Bearsampp/module-consolez) with windows terminal and oh-my-posh
we've just started building our program Bearsampp which is Open Source so we're still very new to all this.  It's a fork from an abandoned project http://neard.io which we're trying to make better.

**Comment by JanDeDobbeleer** [comment]
@N6REJ if you run `oh-my-posh font install`, then select agave, does it work? Not sire if it's an issue with the name, I don't remember including a failsafe in case there's no match.

**Comment by N6REJ** [comment]
@JanDeDobbeleer it hangs @ "installing agave"

**Comment by N6REJ** [comment]
"JetBrainsMono" does same thing

**Comment by N6REJ** [comment]
when I open powershell now I get 
```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

Out-Host : A positional parameter cannot be found that accepts argument 'My'.
At line:1 char:1
+ Oh My Posh Error:
+ ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Out-Host], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.OutHostCommand

Invoke-Expression : Cannot bind argument to parameter 'Command' because it is an empty string.
At D:\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:1 char:82
+ ... nfig "D:\Documents\Development\oh-my-posh\themes" | Invoke-Expression
+                                                         ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:PSObject) [Invoke-Expression], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Microsoft.PowerShell.Commands.Invo
   keExpressionCommand

read : The term 'read' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ read D:\Documents\Development\oh-my-posh\themes: The handle is invali ...
+ ~~~~
    + CategoryInfo          : ObjectNotFound: (read:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Loading personal and system profiles took 2049ms.
PS C:\windows\system32>
```

**Reply by lewis-yeung** [reply]
Have you read through the [docs](https://ohmyposh.dev/docs/installation/customize)? Through the error message, I caught a glimpse of `D:\Documents\Development\oh-my-posh\themes` which is not an valid path to your [theme](https://ohmyposh.dev/docs/themes) config file (JSON/YAML/TOML). A complete init command is like:

```powershell
oh-my-posh init pwsh --config "$env:POSH_THEMES_PATH/jandedobbeleer.omp.json" | Invoke-Expression
```

**Reply by N6REJ** [reply]
ty!  idk how I missed that.  Yes I tried to read thru the docs.  That does gets the colored folder structure working.  Still can't change/set the font.

**Reply by N6REJ** [reply]
nor does it work for cmd

**Reply by N6REJ** [reply]
@lewis-yeung that doesnt' entirely fix things either.. now it is using the default oh-my-posh but not using the theme.
![image](https://user-images.githubusercontent.com/1850089/185984913-75bcf56d-e8f0-4883-878b-da2f218f1b96.png)
```oh-my-posh.exe init pwsh --config "$env:POSH_THEMES_PATH\D:\Documents\Development\oh-my-posh\themes\paradox.omp.json" | Invoke-Expression```

**Reply by lewis-yeung** [reply]
@N6REJ The `$env:POSH_THEMES_PATH\D:\Documents\Development\oh-my-posh\themes\paradox.omp.json` is definitely an invalid path. Try `oh-my-posh.exe init powershell --config "$env:POSH_THEMES_PATH\paradox.omp.json" | Invoke-Expression` instead. The `$env:POSH_THEMES_PATH` will expand to the path where preset themes are located. It was automatically set by the installer since you installed Oh My Posh via Windows Store.

**Reply by lewis-yeung** [reply]
The `oh-my-posh font install` issue is being tracked in #2338. Just be patient and wait for a follow-up fix.

**Reply by lewis-yeung** [reply]
> nor does it work for cmd

@N6REJ What's the exact problem in CMD? Did you follow the [docs](https://ohmyposh.dev/docs/installation/customize)?

**Reply by N6REJ** [reply]
> > nor does it work for cmd
> 
> @N6REJ What's the exact problem in CMD? Did you follow the [docs](https://ohmyposh.dev/docs/installation/customize)?

I believe I did.  It says "load" is not a valid command....
```
Microsoft Windows [Version 10.0.19044.1889]
(c) Microsoft Corporation. All rights reserved.

C:\windows\system32>load(io.popen('oh-my-posh init cmd --config D:\Documents\Development\oh-my-posh\themes\paradox.omp.json'):read("*a"))()
'load' is not recognized as an internal or external command,
operable program or batch file.

C:\windows\system32>
```

**Reply by JanDeDobbeleer** [reply]
@N6REJ you need to install and use clink, and that command needs to be added to the lua file which is read by clink (so not direct). Please follow the [docs](https://ohmyposh.dev/docs/installation/prompt) for cmd to the letter and you will succeed.

**Reply by N6REJ** [reply]
@JanDeDobbeleer ok, I'm closer... both with powershell and cmd I can't use any theme except default.
in cmd its coming out like this. 
![image](https://user-images.githubusercontent.com/1850089/186032419-1d45073d-102f-4dd1-a467-08aeeea2f732.png)

if I point it directly at the file it doesn't show the colored prompts at all and no error

**Comment by N6REJ** [comment]
I don't know if this is helpful but using the included .json's does let it load but doesn't change the prompt.
![image](https://user-images.githubusercontent.com/1850089/186223254-168f1c83-1c89-4e8b-97c4-1277b481bb34.png)

**Reply by JanDeDobbeleer** [reply]
@N6REJ th docs state for cmd:

> Use the full path to the config file, not the relative path.

So tilde won't work there. Also, make sure to install a [font](https://ohmyposh.dev/docs/installation/fonts) as well.

**Reply by N6REJ** [reply]
so far can't get the font to install

**Reply by JanDeDobbeleer** [reply]
@N6REJ what do you mean?

**Reply by N6REJ** [reply]
when I run "oh-my-posh font install" then select a font, it says downloading, then installing and just sits there and spins.  Same if I do it all i one line, like 
![image](https://user-images.githubusercontent.com/1850089/187180192-164bdbe0-2060-470f-8b52-deb2321d7555.png)

**Reply by JanDeDobbeleer** [reply]
@N6REJ also on the latest release? I did a quick update earlier today.

**Reply by N6REJ** [reply]
@JanDeDobbeleer no, it still doesn't work.
![image](https://user-images.githubusercontent.com/1850089/187211032-d7b99c45-64e1-4e54-9511-582c212a6a52.png)

**Reply by JanDeDobbeleer** [reply]
@N6REJ are you admin?

**Reply by N6REJ** [reply]
yes

**Comment by N6REJ** [comment]
also, I've followed the directions as accurately as I know how but it doesn't work for cmd if I put a theme in.  If I use a url then I get the .json as shown previously.  If I put the full path in I get nothing at all like this...
![image](https://user-images.githubusercontent.com/1850089/187212957-16dc3e5b-6864-468e-8a1b-633f6e7ee793.png)
if I don't put anything in I get 
![image](https://user-images.githubusercontent.com/1850089/187213149-93680291-93fe-4535-baa5-5f2cff50eac4.png)

**Reply by JanDeDobbeleer** [reply]
@N6REJ try setting `clink set lua.break_on_error true` and restart cmd to see if there's an error in the script

**Reply by N6REJ** [reply]
with ``load(io.popen('oh-my-posh init cmd --config https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/themes/paradox.omp.json'):read("*a"))()``` it get
```
Microsoft Windows [Version 10.0.19044.1889]
(c) Microsoft Corporation. All rights reserved.

Clink v1.3.40.1901c2
Copyright (c) 2012-2018 Martin Ridgers
Portions Copyright (c) 2020-2022 Christopher Antos
https://github.com/chrisant996/clink

comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:28:46: invalid char escape
comments:429:10: literal not terminated
comments:429:10: invalid char literal
comments:988:25: literal not terminated
comments:992:13: literal not terminated
comments:1039:25: literal not terminated
comments:1043:13: literal not terminated
comments:1660:12: literal not terminated
comments:1660:12: invalid char literal
comments:1684:13: invalid char literal
comments:1699:21: invalid char literal
comments:1700:17: invalid char literal
comments:1700:30: invalid char literal
comments:1701:132: invalid char literal
C:\Program Files (x86)\clink\oh-my-posh.lua:1: attempt to call a nil value
C:\windows\system32>
```
otherwise just a normal screen like I showed above before

**Reply by lewis-yeung** [reply]
> If I put the full path in I get nothing at all like this...
> ![image](https://user-images.githubusercontent.com/1850089/187212957-16dc3e5b-6864-468e-8a1b-633f6e7ee793.png)

@N6REJ In `load(io.popen('oh-my-posh init cmd --config e:\paradox.omp.json'):read("*a"))()`, you should use `\\` or `/` instead of `\` because a backslash (`\`) is used for escaping in an Lua string. So please try `load(io.popen('oh-my-posh init cmd --config e:/paradox.omp.json'):read("*a"))()`. :)

**Reply by N6REJ** [reply]
ok, E:/paradox.omp.json works.  url or full path even with \\ doesn't.

**Reply by lewis-yeung** [reply]
@N6REJ For a URL, you should use `https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/paradox.omp.json` instead of `https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/themes/paradox.omp.json`. 😄

**Comment by N6REJ** [comment]
So, all thats left is getting the font to install
it says its installing and completes, but I don't see a change AND when I select it manually in terminal I get an error
![image](https://user-images.githubusercontent.com/1850089/187579993-dd55ab14-2b3d-40ec-8c52-4877cff58480.png)
with 8.36.5

**Reply by JanDeDobbeleer** [reply]
@N6REJ is the font listed in the font folder? You might need to open it to see the actual name.

**Reply by N6REJ** [reply]
the folder in oh-my-posh?

**Reply by JanDeDobbeleer** [reply]
No, on your device. You can have a look at the fonts installed on your machine, that said, you need to be admin to install a font (and run that command). The latest version will also alert you of that.

**Reply by N6REJ** [reply]
yes, the fonts are installed in windows.  I installed the entire library.

**Reply by JanDeDobbeleer** [reply]
I just did the same and it works (on two machines)

**Reply by N6REJ** [reply]
Did you use powershell or terminal?

**Reply by N6REJ** [reply]
idk if this is helpful, but I just updated, as shown below and one font installed, though I couldn't see a difference, but the other didn't install at all.
![image](https://user-images.githubusercontent.com/1850089/187867127-9551fbad-3b05-4487-af2a-0d37ff3e131b.png)
Should I completely delete and start over?

**Reply by JanDeDobbeleer** [reply]
That could be an issue with that font file, I can check if we can skip incorrect files and not fail on install.

**Reply by N6REJ** [reply]
I feel like I'm being a hassle.  I'm doing my best to do things properly.

**Reply by JanDeDobbeleer** [reply]
@N6REJ not at all. We can only improve things.

**Reply by JanDeDobbeleer** [reply]
hmm that one install without an issue on my machine. Which Windows version are you on?

**Reply by N6REJ** [reply]
![image](https://user-images.githubusercontent.com/1850089/188062517-6b97bee0-ba3c-4ea4-9861-54634f2777ea.png)

**Reply by jwaisner** [reply]
I am also experiencing the problem where the fonts are not loading. I just did a clean install as well.

Checking windows defaults for Powershell fonts, the font is not loaded. I used Meslo and the font install command worked.

![image](https://user-images.githubusercontent.com/12568201/188529480-f8b19616-b969-4ed0-a1e1-4f030ab421c4.png)


OS: Windows 11

![image](https://user-images.githubusercontent.com/12568201/188529290-e1de3139-8a09-4792-892f-c1075c5f548b.png)

**Reply by JanDeDobbeleer** [reply]
@jwaisner dis you restart the terminal?

**Reply by N6REJ** [reply]
yes, we both did a completely clean install with same results.

**Reply by N6REJ** [reply]
idk if this helps but the work around is to manually select which font it should use, in settings, and then its fine.

**Reply by JanDeDobbeleer** [reply]
@N6REJ yes, that's always a step. It only installs a font, we don't set it for you. I'll add that distinction but my assumption was that there's a clear difference between installation and configuration.

**Reply by N6REJ** [reply]
glad we sorted it out.. sorry if I/we were dumb.

**Reply by JanDeDobbeleer** [reply]
@N6REJ nobody's dumb, sorry if my comment made you feel that way. I'll adjust the docs so this is clear.
